exdb/forms.py

```diff
@@ -72,59 +74,62 @@ class Meta:
+            'type').needs_verification if self.cleaned_data.get('type') else False
+        name = "" if not self.cleaned_data.get('type') else self.cleaned_data.get('type').name
+        min_dt = datetime.min.replace(tzinfo=utc)
+        max_dt = datetime.max.replace(tzinfo=utc)
+
+        # conditions format (validation_check, validation_error),
+        conditions = (
+            (not self.cleaned_data.get('description'), ValidationError(_('A description is required'))),
+            (not self.cleaned_data.get('end_datetime'), ValidationError(_('An end time is required'))),
+            (not self.cleaned_data.get('start_datetime'), ValidationError(_('A start time is required'))),
+            (not self.cleaned_data.get('sub_type'), ValidationError(_('The sub type field is required'))),
+            (not self.cleaned_data.get('type'), ValidationError(_('The type field is required'))),
+            (self.cleaned_data.get('type') and not self.approval_form and
+                not self.cleaned_data.get('next_approver') and needs_verification,
+             ValidationError(_('Please select the supervisor to review this experience'))),
+            (self.cleaned_data.get('start_datetime', max_dt) >= self.cleaned_data.get('end_datetime', min_dt),
+             ValidationError(_('Start time must be before end time'))),
+            ((self.cleaned_data.get('type') and not needs_verification) and (self.cleaned_data.get('start_datetime', max_dt) > self.when),
+             ValidationError(_('%(name)s experiences must have happened in the past') % {'name': name})),
+            ((self.cleaned_data.get('type') and not needs_verification) and (not self.cleaned_data.get(
+                'attendance') or self.cleaned_data.get('attendance') < 1),
+             ValidationError(_('%(name)s events must have an attendance') % {'name': name})),
+            (self.cleaned_data.get('type') and not needs_verification and not self.cleaned_data.get('audience'),
+             ValidationError(_('%(name)s events must have an audience') % {'name': name})),
+            (self.cleaned_data.get('type') and needs_verification and self.cleaned_data.get('attendance'),
+             ValidationError(_('%(name)s events cannot have an attendance') % {'name': name})),
+            (self.cleaned_data.get('type') and needs_verification and self.cleaned_data.get('start_datetime', min_dt) < self.when,
+             ValidationError(_('%(name)s events cannot happen in the past') % {'name': name})),
+            (needs_verification and self.cleaned_data.get('next_approver')
+                and not self.cleaned_data.get('next_approver').groups.filter(name='hs').exists(),
+             ValidationError(_('Supervisor must have permissions to approve and deny experiences'))),
+            (not needs_verification and not self.cleaned_data.get('conclusion'),
+             ValidationError(_('%(name)s events must have a conclusion') % {'name': name})),
```

Comment:
Submitting an empty experience gives this error, Without showing the user the conclusion field. Not really a big error, just an observation